lib/pacemaker/pcmk_resource.c

```diff
@@ -103,6 +104,68 @@ best_op(const pcmk_resource_t *rsc, const pcmk_node_t *node)
     return best;
 }
 
+/*!
+ * \internal
+ * \brief Remove a resource
+ *
+ * \param[in,out] cib       An open connection to the CIB
+ * \param[in]     cib_opts  Options to use in the CIB operation call
+ * \param[in]     rsc_id    Resource to remove
+ * \param[in]     rsc_type  Type of the resource ("primitive", "group", etc.)
+ *
+ * \return Standard Pacemaker return code
+ */
+int
+pcmk__resource_delete(cib_t *cib, uint32_t cib_opts, const char *rsc_id,
+                      const char *rsc_type)
+{
+    int rc = pcmk_rc_ok;
+    xmlNode *msg_data = NULL;
+
+    if (cib == NULL) {
+        return ENOTCONN;
```

Comment:
FYI we could open and close a local connection within this function (like the public one) when cib==NULL, but until we have a use case there's no need

Reply:
At the moment, the private functions are a delightful mix of expecting an already open cib connection (pcmk__status) and creating their own connection without taking it as a parameter (pcmk__simulate).  I can do whatever here, but I feel like we should standardize the behavior across all existing private functions so we have less to think about later.

Reply:
If we want common behavior, then passing the cib connection and opening a local one if it's NULL makes the most sense. I wouldn't worry about it with this PR though, we can add it separately